### ISSUE TITLE
Issue when publishing stations

### ISSUE THREAD

**Issue by davidpodeur** [wmo-im/wis2box#633]
Hi,

I am running a fresh WIS2BOX 1.0b6 install on Ubuntu 22.04.4 and I am not able to publish/display stations.

For debugging purposes, I have increase the verbosity:
```
wis2box@wis2node-fra-int:~/wis2box-1.0b6$ grep LOG /home/wis2box/wis2box-1.0b6/wis2box.env
WIS2BOX_LOGGING_LOGLEVEL=INFO
WIS2BOX_LOGGING_LOGFILE=stdout
```
Here is my station file, which I have reduced to two lines (one from the examples, and one for my centre-id):

```
root@ac0c11f44ce6:/home/wis2box# cat /data/wis2box/metadata/station/station_list.csv
station_name,wigos_station_identifier,traditional_station_identifier,facility_type,latitude,longitude,elevation,barometer_height,territory_name,wmo_region
BALAKA,0-454-2-AWSBALAKA,AWSBALAKA,Land (fixed),-14.983333,34.966666,618,,Malawi,I
ABBEVILLE,0-20000-0-07005,07005,Land (fixed),50.136000,1.834000,69,,France,VI
```

When ingesting:

```
root@ac0c11f44ce6:/home/wis2box# wis2box metadata station publish-collection
[2024-02-29T08:02:44Z] {/usr/local/lib/python3.8/dist-packages/elastic_transport/_transport.py:335} INFO - DELETE http://elasticsearch:9200/stations/_doc/0-454-2-AWSBALAKA [status:200 duration:0.018s]
[2024-02-29T08:02:44Z] {/usr/local/lib/python3.8/dist-packages/elastic_transport/_transport.py:335} INFO - HEAD http://elasticsearch:9200/stations [status:200 duration:0.002s]
[2024-02-29T08:02:44Z] {/usr/local/lib/python3.8/dist-packages/elastic_transport/_transport.py:335} INFO - PUT http://elasticsearch:9200/_bulk [status:200 duration:0.011s]
[2024-02-29T08:02:44Z] {/usr/local/lib/python3.8/dist-packages/elastic_transport/_transport.py:335} INFO - DELETE http://elasticsearch:9200/stations/_doc/0-20000-0-07005 [status:200 duration:0.051s]
[2024-02-29T08:02:44Z] {/usr/local/lib/python3.8/dist-packages/elastic_transport/_transport.py:335} INFO - HEAD http://elasticsearch:9200/stations [status:200 duration:0.002s]
[2024-02-29T08:02:44Z] {/usr/local/lib/python3.8/dist-packages/elastic_transport/_transport.py:335} INFO - PUT http://elasticsearch:9200/_bulk [status:200 duration:0.045s]
[2024-02-29T08:02:44Z] {/app/wis2box/metadata/station.py:381} INFO - Updated station list: ['0-454-2-AWSBALAKA', '0-20000-0-07005']
Done
```

However, my station list remains blank:
![image](https://github.com/wmo-im/wis2box/assets/39593874/97eeb074-97e8-466f-bd22-1d60bdc34516)

And if I click on "add station", I get "HTTP error fetching code list, please see console."
![image](https://github.com/wmo-im/wis2box/assets/39593874/22bd2aea-bd3e-4a01-871f-adf511c930e8)

=> Which console ?

I also have this issue, which might be related:
```
wis2box@wis2node-fra-int:~/wis2box-1.0b6$ docker compose logs --tail=10 --follow
WARN[0000] The "WIS2BOX_HOST_DATADIR" variable is not set. Defaulting to a blank string.
invalid spec: :/data/wis2box:rw: empty section between colons
```
```
wis2box@wis2node-fra-int:~$ grep WIS2BOX_HOST_DATADIR ./wis2box-1.0b6/wis2box.env
WIS2BOX_HOST_DATADIR=/home/wis2box/wis2box-data
wis2box@wis2node-fra-int:~$ grep "/data/wis2box" ./wis2box-1.0b6/wis2box.env
WIS2BOX_DATADIR=/data/wis2box
```

Is is an station metadata ingestion issue or a metadata fetch issue ?
 
Any help would be much appreciated.

David

**Comment by maaikelimper** [comment]
Hi David,

The error "The "WIS2BOX_HOST_DATADIR" variable is not set. Defaulting to a blank string." indicates a configuration issue, your wis2box.env needs to be pre-populated with configuration-values for the wis2box-stack to start correctly.

I would recommend stopping the wis2box, pruning the volume (clean install) and start by using the configuration-script to create you wis2box.env

Stop wis2box:
python3 wis2box-ctl.py stop

Prune volumes:
docker volume prune -a

Run configuration script:
python3 wis2box-create-config.py

For more information, see the documentation for Installation and Configuration:
https://docs.wis2box.wis.wmo.int/en/latest/user/setup.html

**Comment by davidpodeur** [comment]
Thanks for the hint.

There was indeed a mistake in my wis2box.env (not generated by `wis2box-create-config.py`).

The entries `WIS2BOX_URL` and `WIS2BOX_API_URL` were lacking the `http://` prefix.

**Comment by davidpodeur** [comment]
The issue with  WIS2BOX_HOST_DATADIR remained:
```
wis2box@wis2node-fra-int:~/wis2box-1.0b6$ docker compose logs --tail=10 --follow
WARN[0000] The "WIS2BOX_HOST_DATADIR" variable is not set. Defaulting to a blank string.
invalid spec: :/data/wis2box:rw: empty section between colons
```

In order to fix, I had to remove the reference to $WIS2BOX_HOST_DATADIR  in the docker-compose.yml file.

Not working:
```
wis2box@wis2node-fra-int:~/wis2box-1.0b6$ grep -A 12 "  wis2box-management:" docker-compose.yml
  wis2box-management:
    container_name: wis2box-management
    mem_limit: 1g
    memswap_limit: 1g
    restart: always
    image: ghcr.io/wmo-im/wis2box-management:1.0b6
    env_file:
      - wis2box.env
    volumes:
      # This does not work
      - ${WIS2BOX_HOST_DATADIR}:/data/wis2box:rw
      # Trying without the variable
      #- /home/wis2box/wis2box-data:/data/wis2box:rw

wis2box@wis2node-fra-int:~/wis2box-1.0b6$ python3 wis2box-ctl.py start
Error response from daemon: plugin loki:latest already exists
Error response from daemon: plugin already enabled: plugin loki:latest is enabled
[+] Running 13/15
 ⠋ Network wis2box_project_default   Created                                                                                                                                                           119.0s
 ⠏ Network wis2box_project_vpcbr     Created                                                                                                                                                           118.9s
 ✔ Container elasticsearch           Healthy                                                                                                                                                            69.2s
 ✔ Container loki                    Started                                                                                                                                                            19.1s
 ✔ Container mosquitto               Started                                                                                                                                                            14.6s
 ✔ Container prometheus              Started                                                                                                                                                            15.4s
 ✔ Container grafana                 Started                                                                                                                                                            11.7s
 ✔ Container wis2box-webapp          Started                                                                                                                                                            12.3s
 ✔ Container wis2box-api             Healthy                                                                                                                                                           110.9s
 ✔ Container wis2box-minio           Healthy                                                                                                                                                            69.7s
 ✔ Container wis2box-ui              Started                                                                                                                                                            70.0s
 ✔ Container wis2box-management      Started                                                                                                                                                           110.6s
 ✔ Container nginx                   Started                                                                                                                                                            70.0s
 ✔ Container mqtt_metrics_collector  Started                                                                                                                                                           112.6s
 ✔ Container wis2box-auth            Started                                                                                                                                                           112.5s
wis2box@wis2node-fra-int:~/wis2box-1.0b6$ docker compose logs
WARN[0000] The "WIS2BOX_HOST_DATADIR" variable is not set. Defaulting to a blank string.
invalid spec: :/data/wis2box:rw: empty section between colons
```

This works:

```
wis2box@wis2node-fra-int:~/wis2box-1.0b6$ grep -A 12 "  wis2box-management:" docker-compose.yml
  wis2box-management:
    container_name: wis2box-management
    mem_limit: 1g
    memswap_limit: 1g
    restart: always
    image: ghcr.io/wmo-im/wis2box-management:1.0b6
    env_file:
      - wis2box.env
    volumes:
      # This does not work
      #- ${WIS2BOX_HOST_DATADIR}:/data/wis2box:rw
      # Trying without the variable
      - /home/wis2box/wis2box-data:/data/wis2box:rw

wis2box@wis2node-fra-int:~/wis2box-1.0b6$ python3 wis2box-ctl.py restart
[+] Running 15/15
 ✔ Container grafana                 Removed                                                                                                                                                             1.4s
 ✔ Container prometheus              Removed                                                                                                                                                             2.7s
 ✔ Container mqtt_metrics_collector  Removed                                                                                                                                                            11.6s
 ✔ Container nginx                   Removed                                                                                                                                                             3.4s
 ✔ Container wis2box-auth            Removed                                                                                                                                                             3.4s
 ✔ Container wis2box-webapp          Removed                                                                                                                                                            12.3s
 ✔ Container loki                    Removed                                                                                                                                                             4.6s
 ✔ Container wis2box-ui              Removed                                                                                                                                                             1.1s
 ✔ Container wis2box-management      Removed                                                                                                                                                            11.5s
 ✔ Container wis2box-minio           Removed                                                                                                                                                             5.9s
 ✔ Container wis2box-api             Removed                                                                                                                                                            11.7s
 ✔ Container mosquitto               Removed                                                                                                                                                            14.1s
 ✔ Container elasticsearch           Removed                                                                                                                                                             5.8s
 ✔ Network wis2box_project_vpcbr     Removed                                                                                                                                                             0.3s
 ✔ Network wis2box_project_default   Removed                                                                                                                                                             0.6s
[+] Running 13/15
 ⠹ Network wis2box_project_vpcbr     Created                                                                                                                                                           109.3s
 ⠙ Network wis2box_project_default   Created                                                                                                                                                           109.2s
 ✔ Container prometheus              Started                                                                                                                                                            17.1s
 ✔ Container grafana                 Started                                                                                                                                                            21.5s
 ✔ Container wis2box-webapp          Started                                                                                                                                                            18.1s
 ✔ Container loki                    Started                                                                                                                                                            23.6s
 ✔ Container elasticsearch           Healthy                                                                                                                                                            58.4s
 ✔ Container mosquitto               Started                                                                                                                                                            13.7s
 ✔ Container wis2box-api             Healthy                                                                                                                                                            97.4s
 ✔ Container wis2box-minio           Healthy                                                                                                                                                            56.3s
 ✔ Container wis2box-ui              Started                                                                                                                                                            56.4s
 ✔ Container wis2box-management      Started                                                                                                                                                            98.6s
 ✔ Container nginx                   Started                                                                                                                                                            59.4s
 ✔ Container wis2box-auth            Started                                                                                                                                                           100.4s
 ✔ Container mqtt_metrics_collector  Started                                                                                                                                                           100.6s

wis2box@wis2node-fra-int:~/wis2box-1.0b6$ docker compose logs
wis2box@wis2node-fra-int:~/wis2box-1.0b6$
```

This is possibly an issue with the docker compose version.

I am using:
```
wis2box@wis2node-fra-int:~/wis2box-1.0b6$ docker compose version
Docker Compose version v2.24.6
```

Anyway, I now manage to publish data, get MQTT notifications and download corresponding data files.


One more question, it is normal that I do not have the usual map on the main page ? (I used to have it in 1.0b2)

![image](https://github.com/wmo-im/wis2box/assets/39593874/8b871620-b466-472e-8970-346ac49e9393)

**Comment by maaikelimper** [comment]
Hi David,
The absence of data in the wis2box-ui could mean that the ui did not find discovery metadata in your WIS2BOX_API. Can you check if you see discovery metadata under <your-host>/oapi/collections/discovery-metadata/items ?
Another thing to check is the console that is available in most browsers by right-clicking on the page and selecting "inspect": 

![image](https://github.com/wmo-im/wis2box/assets/19169339/3522cc6b-51e5-4711-a989-b8a1a8297e27)

![image](https://github.com/wmo-im/wis2box/assets/19169339/1b04d63b-a2ce-47bb-bfc1-cf066cd94338)

The console might indicate that it is unable to find your API-endpoint ?

I will try to validate the issue you reported on the WIS2BOX_HOST_DATADIR variable not being picked up by the docker-compose.yml and see if this is due to your docker compose version, I have not seen this issue before ...

**Comment by maaikelimper** [comment]
I noted that the WIS2BOX_HOST_DATADIR error is always printed when running `docker compose logs` yet the wis2box-management gets correctly intialized using the WIS2BOX_HOST_DATADIR from the wis2box.env :
![image](https://github.com/wmo-im/wis2box/assets/19169339/fbb5d7e6-8f9f-474a-a010-26d419f806bf)

The error appears to be a result of the "docker compose" command running without specifying the --env-file :
![image](https://github.com/wmo-im/wis2box/assets/19169339/8992a13f-b0e1-4f50-9823-afc7db62c239)

(note that within the wis2box-ctl.py --env-file=wis2box.env is added to all docker compose commands)

**Comment by maaikelimper** [comment]
David, my apologies, I sent the link to the 'latest' documentation which refers to commands not available in wis2box-1.0b6

Please use the commands documented here:
https://docs.wis2box.wis.wmo.int/en/1.0b6/user/setup.html#runtime-configuration
